### PR TITLE
groom zcode API

### PR DESCRIPTION
**Pull request by nwt**
I've tried to improve some names in package zcode that didn't feel quite right.

(I'll rebase this onto master after #196 is merged.)

### CODE REVIEW

**Comment by nwt** [comment]
I agree that "simple value" isn't great. I don't love "scalar" because it makes me think of numbers, but I can get past that if others like it.

**Comment by aswan** [comment]
Just spitballing...  what about "primitive types"?

**Comment by nwt** [comment]
Sold!

**Comment by mccanne** [comment]
The Avro spec uses "primitive" and "complex".   Wherever we land, we should update the ZNG spec for consistency.

**Comment by nwt** [comment]
@aswan, @mccanne, @henridf, @mattnibs: Do any of you have a preference between container and complex? I don't right now.

**Comment by henridf** [comment]
"complex" makes me think of complex numbers so my vote would be for container. 
And if [composite](https://en.wikipedia.org/wiki/Composite_data_type) were on the ballot, my vote would be for that (and/or compound) over container.

**Comment by aswan** [comment]
Of the options that have come up so far, I find container most descriptive.  It also means less churn since we're already using it.  Consider this a soft preference for just sticking with it...

File: zcode/iter.go [hunk 1]
```diff
@@ -4,20 +4,21 @@ import (
...
 func (i *Iter) Done() bool {
 	return len(*i) == 0
 }
 
-// Next returns the next value as a Bytes type.  It returns an empty slice for
-// an empty or zero-length value and nil for an unset value.
+// Next returns the body of the next value along with a boolean that is true if
+// the value is a container.  It returns an empty slice for an empty or
+// zero-length value and nil for an unset value.
 func (i *Iter) Next() (Bytes, bool, error) {
```
Comment (mattnibs) [inline zcode/iter.go]: You don't have to do this here but I think this api should return ([]byte, Bytes, error). The first return value being the raw bytes of a primitive, the second being a container... Just a thought.

**Comment by nwt** [comment]
I do like @henridf's suggestions (composite and compound), but I also like container. I'm inclined to stick with it for the reason @aswan mentioned.

**Comment by nwt** [comment]
Other packages still refer to scalar and compound values. To keep this focused, I'll update them in another PR.

**Comment by nwt** [comment]
@aswan, @mccanne, @henridf, @mattnibs: Please take another look.

**Pull request merged**